### PR TITLE
Make IRecursivePatternOperation public

### LINKED ISSUE #35843: Make IRecursivePatternOperation public
**Version Used**: 
3.0.0

**Description**
Right now, there is no way to analyze the operation behind the pattern matching code like this: `a is (var b, var c)`  without using Reflection. Please make `IRecursivePatternOperation` interface public.

### PR DESCRIPTION
**Pull request by gafter**
Fixes #35843

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Awesome thanks!  i can remove some ugly reflection code now :)

**Comment by 333fred** [comment]
>         RecursivePattern = 0x67,

Wait, this was public? Oops.

---
Refers to: src/Compilers/Core/Portable/Operations/OperationKind.cs:237 in a84dc72. [](commit_id = a84dc72ccbaad738dcde01bc1f2ed70d5fbc3ade, deletion_comment = False)

File: src/Compilers/Core/Portable/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -34,6 +34,16 @@ Microsoft.CodeAnalysis.NullableFlowState
 Microsoft.CodeAnalysis.NullableFlowState.MaybeNull = 2 -> Microsoft.CodeAnalysis.NullableFlowState
 Microsoft.CodeAnalysis.NullableFlowState.NotApplicable = 0 -> Microsoft.CodeAnalysis.NullableFlowState
 Microsoft.CodeAnalysis.NullableFlowState.NotNull = 1 -> Microsoft.CodeAnalysis.NullableFlowState
+Microsoft.CodeAnalysis.OperationKind.PropertySubpattern = 107 -> Microsoft.CodeAnalysis.OperationKind
+Microsoft.CodeAnalysis.Operations.IPropertySubpatternOperation
+Microsoft.CodeAnalysis.Operations.IPropertySubpatternOperation.Member.get -> Microsoft.CodeAnalysis.IOperation
+Microsoft.CodeAnalysis.Operations.IPropertySubpatternOperation.Pattern.get -> Microsoft.CodeAnalysis.Operations.IPatternOperation
+Microsoft.CodeAnalysis.Operations.IRecursivePatternOperation
```
Comment (mavasani) [inline src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: We should remove this IDE reflection based hack that was added for this API: http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Extensions/OperationExtensions.cs,64

Comment (mavasani) [reply src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: `case IOperation iop when iop.GetType().GetInterfaces().Any(i => i.Name == "IRecursivePatternOperation"):`

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review.

**Comment by gafter** [comment]
@dotnet/roslyn-compiler May I please have a second review from the compiler team?

**Comment by jcouv** [comment]
Looking

**Pull request merged**